docs/js/immutablejs.md

```diff
@@ -11,66 +11,78 @@ for a good explanation of the more intricate benefits it has.
 
 In our reducers, we make the initial state an immutable data structure with the
 `fromJS` function. We pass it an object or an array, and it takes care of
-converting it to a compatible one. (Note: the conversion is performed deeply so
-that even arbitrarily nested arrays/objects are immutable stuctures too!)
+converting it to a immutable data structure. (Note: the conversion is performed deeply so
+that even arbitrarily nested arrays/objects are immutable structures too!)
 
 ```JS
 import { fromJS } from 'immutable';
 
 const initialState = fromJS({
-  myData: 'Hello World!',
+  myData: {
+  	message: 'Hello World!'
+  },
 });
 ```
 
-To react to an incoming actions our reducers can use the `.set` and the `.setIn`
-functions.
+When actions are dispatched and our reducers need to set new state they can do so by using setter methods such as [`.set`](https://facebook.github.io/immutable-js/docs/#/Map/set) and [`.update`](https://facebook.github.io/immutable-js/docs/#/Map/update).  
```

Comment:
I'd prefer it if we said something like "our reducers need to _return_ the new state"